lib/foreman_tasks_core/runner/parent.rb

```diff
@@ -0,0 +1,52 @@
+module ForemanTasksCore
+  module Runner
+    class Parent < Base
+
+      # targets = { hostname => { :execution_plan_id => "...", :run_step_id => id,
+      #                           :input => { ... } }
+      def initialize(targets = {})
+        @targets = targets
+        super()
+      end
+
+      def generate_updates
+        @outputs.reduce({}) do |acc, (key, value)|
+          if value.empty? && @exit_status.nil?
+            acc
+          else
+            @outputs[key] = ForemanTasksCore::ContinuousOutput.new
+            key = host_action(key) unless key == :control
```

Comment:
This reads kind of hard, as there is no explanation what the `:control` key means, and it's kind of strange that we mix the `:control` key, with some `Dynflow::Action::Suspended` objects. It would be nice if you could think of possible refactoring to help with readability.